README.md

```diff
@@ -25,34 +37,72 @@ Example installation:
+- HPX and DLAF must have a compatible `CMAKE_BUILD_TYPE`: they must be built both in Debug, or with any combination of release types (Release, RelWithDebInfo or MinSizeRel)
+
+And here the main CMake options for DLAF build customization:
+
+CMake option | Values | Note
+:---|:---|:---
+`HPX_DIR` | CMAKE:PATH | Location of the HPX CMake-config file
+`blaspp_DIR` | CMAKE:PATH | Location of the blaspp CMake-config file
+`lapackpp_DIR` | CMAKE:PATH | Location of the lapackpp CMake-config file
+`DLAF_WITH_MKL` | `{ON,OFF}` (default: `OFF`) | if blaspp/lapackpp is built with MKL
+`MKL_ROOT` | CMAKE:PATH | Location of the MKL library
+`DLAF_ASSERT_ENABLE` | `{ON,OFF}` (default: `ON`) | enable/disable cheap assertions
+`DLAF_ASSERT_MODERATE_ENABLE` | `{ON,OFF}` (default: `ON` in Debug, `OFF` otherwise) | enable/disable moderate assertions
+`DLAF_ASSERT_HEAVY_ENABLE` | `{ON,OFF}` (default: `ON` in Debug, `OFF` otherwise) | enable/disable heavy assertions
+`DLAF_WITH_CUDA` | `{ON,OFF}` (default: `OFF`) | enable CUDA support
+`DLAF_BUILD_MINIAPPS` | `{ON,OFF}` (default: `ON`) | enable/disable building miniapps
+`DLAF_WITH_TEST` | `{ON,OFF}` (default: `ON`) | enable/disable building tests
+`DLAF_INSTALL_TESTS` | `{ON,OFF}` (default: `OFF`) | enable/disable installing tests
+`DLAF_MPI_PRESET` | `{plain-mpi, slurm, custom}` (default `plain-mpi`) | presets for MPI configuration for tests. See [CMake Doc](https://cmake.org/cmake/help/latest/module/FindMPI.html?highlight=mpiexec_executable#usage-of-mpiexec) for additional information
+`DLAF_TEST_RUNALL_WITH_MPIEXEC` | `{ON, OFF}` (default: `OFF`) | Use mpi runner also for non-MPI based tests
+`DLAF_HPXTEST_EXTRA_ARGS` | CMAKE:STRING | Additional HPX command-line options for tests
+`BUILD_DOC` | `{ON,OFF}` (default: `OFF`) | enable/disable documentation generation
+
+### Link your program/library with DLAF
 
 Using DLAF in a CMake project is extremely easy!
 
-Let's use the variable `$DLAF_ROOT` for referring to the install path of DLAF.
-
-Configure your project with:
+In the following, the variable `DLAF_INSTALL_PREFIX` is set to where DLAF is installed. In case you used spack for installing DLAF, you can easily set it with:
 
 ```bash
-cmake -DDLAF_DIR="$DLAF_ROOT/lib/cmake" ..
+export DLAF_INSTALL_PREFIX=`spack location -i dla-future`
 ```
 
-Then, it is just as simple as:
+Then, you can configure your project with one of the following:
+
+```bash
+# By appending the value to the CMAKE_INSTALL_PREFIX
+cmake -DCMAKE_INSTALL_PREFIX=${DLAF_INSTALL_PREFIX} ..
 
+# ... or by setting DLAF_DIR
+cmake -DDLAF_DIR="$DLAF_INSTALL_PREFIX/lib/cmake" ..
 ```
-find_package(DLAF)
 
-# ...
+Then, it is just as simple as adding these directives in your `CMakeLists.txt`:
 
-target_link_libraries(<your_target> PRIVATE DLAF)
+```
+find_package(DLAF)
+# ... and then for your executable/library target
+target_link_libraries(<your_target> PRIVATE DLAF::DLAF)
 ```
```

Comment:
I was thinking that it may be useful to show an example of a "real code" linked to DLAF. What about using miniapp?